compiler/rustc_typeck/src/coherence/orphan.rs

```diff
@@ -243,13 +245,25 @@ fn emit_orphan_check_error<'tcx>(
                     _ => ty,
                 };
                 let this = "this".to_string();
-                let (ty, postfix) = match &ty.kind() {
+                let (ty_str, postfix) = match &ty.kind() {
                     ty::Slice(_) => (this, " because slices are always foreign"),
                     ty::Array(..) => (this, " because arrays are always foreign"),
                     ty::Tuple(..) => (this, " because tuples are always foreign"),
                     _ => (format!("`{}`", ty), ""),
                 };
-                let msg = format!("{} is not defined in the current crate{}", ty, postfix);
+                let msg = match &ty.kind() {
+                    ty::RawPtr(ptr_ty) => {
+                        emit_orphan_check_for_impl_on_types(
+                            full_impl_span,
+                            self_ty_span,
+                            ptr_ty,
+                            &mut err,
+                        );
+                        format!("{} type does not meet the coherence rules", ty_str)
```

Comment:
this should be more specific, for example:

`raw pointers are never considered local to the current crate`

or something like that

Reply:
I have no better proposal, lets go with yours

Reply:
In this case, do you want to remove also the formatting string? or do you mean to have an error message like this  `*mut LocalType raw pointers are never considered local to the current crate`